### ISSUE TITLE
@load_dependencies() doesn't work as advertised

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/BinDeps.jl#196]
README.md says that:
> This will make all your libraries available as variables named by the names you gave the dependency. E.g. if you declared a dependency as
```
    library_dependency("libfoo")
```
The libfoo variable will now contain a reference to that library that may be passed to ccall or similar functions. 

But this doesn't appear to work that way. For example, with the file at https://github.com/nalimilan/iconv.jl/blob/f0ce5cd88921d2aa8b4fabf95a100271872e2e4f/deps/build.jl, if I call `@BinDeps.load_dependencies` from `src/iconv.jl`, I get:
```julie
julia> iconv.libiconv
1-element Array{Any,1}:
 ((System Paths,Dict{Any,Any}()),"/lib64/libc.so.6")
```
which cannot be passed directly to `ccall`.

Actually, all the projects using BinDeps I found load `../deps/deps.jl` directly instead of using `@load_dependencies`. This is no ideal, since they need to replicate the checks about the existence of the file and the error message. And it's particularly bad with regard to the docs, as there is apparently no reference to get this to work. Is this behavior a mere bug?

**Comment by tkelman** [comment]
So I think the reason `@load_dependencies` is effectively deprecated is to avoid a runtime dependency on BinDeps. With package precompilation the load-time cost of that may be low enough to not matter any more. That doesn't necessarily mean `@load_dependencies` works properly though.

**Comment by nalimilan** [comment]
I hadn't thought about that. Maybe saving a few tenths of seconds is worth it, though. I would be fine with recommending `include(join.path(".., "deps", "deps.jl")`, but we should update the docs and deprecate `@load_dependencies`.

**Comment by tkelman** [comment]
It is generally nicer to check for the existence of deps.jl first to give a more specific error message if it wasn't created yet. Maybe the load_dependencies stuff would handle that, if it worked.

**Comment by nalimilan** [comment]
Yes, we should at least give an additional line in the examples which prints an error message when the file doesn't exist.

**Comment by Gnimuc** [comment]
Any progress on this and #100?
